### PR TITLE
check connection and blackout perior sending messages && retry encrypt database

### PR DESCRIPTION
**Pull request by b2mdevelopment**
Check connection availability and blackout perior to read messages from sqlite
retry to check database encryption and encrypt it within 20 second and guard against any database exception

### CODE REVIEW

File: src/reyna.Facts/GivenAForwardService.cs [hunk 1]
```diff
@@ -83,6 +102,44 @@ public void WhenCallingStartAndMessageAddedShouldCallPostOnHttpClient()
...
 
         [Fact]
+        public void WhenCallingStartAndItIsBlackoutShouldNotCallPostOnHttpClientOrDeleteMessage()
+        {
+            var networkInterface = new NetworkInterface();
+            networkInterface.CurrentIpAddress = new IPAddress(42);
+            networkInterface.Name = "cellular line";
+            var from = new Time();
+            var to = new Time(from.MinuteOfDay + 1);
+            this.Preferences.SetCellularDataBlackout(new TimeRange(from, to));
```
Comment (charlierobson) [inline src/reyna.Facts/GivenAForwardService.cs:112]: this might be a little fragile - if time is about to change hh:mm:ss:999 for example

**Comment by charlierobson** [comment]
:+1: one comment about a potentially fragile test but it's not a stopper.

File: src/reyna.Facts/GivenAForwardService.cs [hunk 2]
```diff
@@ -2,21 +2,38 @@
...
         public GivenAForwardService()
         {
+            NetworkInterface.NetworkInterfaces = new INetworkInterface[0];
+            this.Preferences = new Preferences();
+            this.Preferences.ResetCellularDataBlackout();
+            this.Preferences.ResetWlanBlackoutRange();
+            this.Preferences.ResetWwanBlackoutRange();
+            this.Preferences.ResetRoamingBlackout();
+            this.Preferences.ResetOnChargeBlackout();
+            this.Preferences.ResetOffChargeBlackout();
```
Comment (b2mdevelopment) [inline src/reyna.Facts/GivenAForwardService.cs]: Can preferences be mocked? @CharlesGarth

File: src/reyna.Facts/GivenAForwardService.cs [hunk 3]
```diff
@@ -83,6 +102,44 @@ public void WhenCallingStartAndMessageAddedShouldCallPostOnHttpClient()
...
+            var from = new Time();
+            var to = new Time(from.MinuteOfDay + 1);
+            this.Preferences.SetCellularDataBlackout(new TimeRange(from, to));
+            NetworkInterface.NetworkInterfaces = new INetworkInterface[] { networkInterface };
+
+            var message = this.CreateMessage();
+            this.ForwardService.Start();
+
+            this.PersistentStore.Add(message);
+            Thread.Sleep(200);
```
Comment (CharlesGarth) [inline src/reyna.Facts/GivenAForwardService.cs:119]: Thread sleeps in unit tests smell a bit can we refactor to remove these?

File: src/reyna.Facts/GivenAForwardService.cs [hunk 4]
```diff
@@ -2,21 +2,38 @@
...
+    using OpenNETCF.Net.NetworkInformation;
     using Reyna.Interfaces;
     using Xunit;
 
     public class GivenAForwardService
     {
         public GivenAForwardService()
         {
+            NetworkInterface.NetworkInterfaces = new INetworkInterface[0];
+            this.Preferences = new Preferences();
```
Comment (CharlesGarth) [inline src/reyna.Facts/GivenAForwardService.cs:18]: Can preferences be mocked?

Comment (b2mdevelopment) [inline src/reyna.Facts/GivenAForwardService.cs:18]: The issue regarding these that, I need to test the main logic without breaking it into smaller,, protected function.
So I used thread sleep to simulate message added to sore and try to send it once added.
Can lead to false alert test? but still better to test full scenario.

**Pull request merged**